### ISSUE TITLE
Bundle runtime dependencies?

### ISSUE THREAD

**Issue by oliverchang** [google/centipede#224]
It looks like currently, centipede relies on:

- libssl
- objdump 
- anything else? 

Would it make sense to vendor these and output these binaries as part of the `bazel build`? (or at least statically link libssl).

This would minimize hidden runtime dependencies and make integration into fuzzing infra a bit nicer, without having to require us to install additional dependencies in the runtime environment that end up causing issues due to version mismatches later.

**Comment by oliverchang** [comment]
I think this does the trick to statically link against the system libssl:

```
diff --git a/BUILD b/BUILD
index 5a2e97d..6032914 100644
--- a/BUILD
+++ b/BUILD
@@ -77,7 +77,7 @@ cc_library(
         "util.cc",
     ],
     hdrs = ["util.h"],
-    linkopts = ["-lcrypto"],
+    linkopts = ["-Wl,-Bstatic -lcrypto -Wl,-Bdynamic -ldl"],
     deps = [
         ":defs",
         ":feature",
```

Which should unblock our immediate needs, but the question remains for if we should build and bundle objdump.

**Comment by kcc** [comment]
`inkopts = ["-Wl,-Bstatic -lcrypto -Wl,-Bdynamic -ldl"],` LGTM

objdump is not used for the main functionality, is it?
`GetPcTableFromBinaryWithTracePC()` where it is used is a special function that is needed for special targets, not part of the usual workflow. But since there is a test for `GetPcTableFromBinaryWithTracePC()`, objdump is needed for testing Centipede.

**Comment by oliverchang** [comment]
> `inkopts = ["-Wl,-Bstatic -lcrypto -Wl,-Bdynamic -ldl"],` LGTM
> 
> objdump is not used for the main functionality, is it? `GetPcTableFromBinaryWithTracePC()` where it is used is a special function that is needed for special targets, not part of the usual workflow. But since there is a test for `GetPcTableFromBinaryWithTracePC()`, objdump is needed for testing Centipede.

Got it, I didn't look too closely at what it was used for. This is only used for binaries that aren't compiled with the correct set of flags (i.e. no pc-table), right? 

In that case, just linking libssl statically should suffice. 

Thanks!

**Comment by ussuri** [comment]
Thanks for the suggested fix. Verified locally: it works. Submitting https://github.com/google/centipede/pull/234 now.
